source/uwp/Renderer/lib/XamlHelpers.cpp

```diff
@@ -359,44 +359,48 @@ namespace AdaptiveCards::Rendering::Uwp::XamlHelpers
-        // image element and then build that into xaml and apply to the root.
-        ComPtr<IAdaptiveImage> adaptiveImage = XamlHelpers::CreateABIClass<IAdaptiveImage>(
-            HStringReference(RuntimeClass_AdaptiveCards_ObjectModel_Uwp_AdaptiveImage));
+        ComPtr<IImage> image = XamlHelpers::CreateABIClass<IImage>(HStringReference(RuntimeClass_Windows_UI_Xaml_Controls_Image));
 
-        HString url;
-        THROW_IF_FAILED(backgroundImage->get_Url(url.GetAddressOf()));
-        THROW_IF_FAILED(adaptiveImage->put_Url(url.Get()));
+        ComPtr<IDependencyObject> imageAsDependencyObject;
+        THROW_IF_FAILED(image.As(&imageAsDependencyObject));
 
-        ComPtr<IAdaptiveCardElement> adaptiveCardElement;
-        THROW_IF_FAILED(adaptiveImage.As(&adaptiveCardElement));
+        ComPtr<IAdaptiveHostConfig> hostConfig;
+        THROW_IF_FAILED(renderContext->get_HostConfig(&hostConfig));
 
-        ComPtr<IAdaptiveElementRendererRegistration> elementRenderers;
-        THROW_IF_FAILED(renderContext->get_ElementRenderers(&elementRenderers));
+        ComPtr<IUriRuntimeClass> imageUrl;
+        GetUrlFromString(hostConfig.Get(), url, imageUrl.GetAddressOf());
 
-        ComPtr<IAdaptiveElementRenderer> elementRenderer;
-        THROW_IF_FAILED(elementRenderers->Get(HStringReference(L"Image").Get(), &elementRenderer));
+        ComPtr<IBitmapImage> bitmapImage =
+            XamlHelpers::CreateABIClass<IBitmapImage>(HStringReference(RuntimeClass_Windows_UI_Xaml_Media_Imaging_BitmapImage));
+        THROW_IF_FAILED(bitmapImage->put_UriSource(imageUrl.Get()));
 
-        ComPtr<IUIElement> background;
-        if (elementRenderer != nullptr)
-        {
-            elementRenderer->Render(adaptiveCardElement.Get(), renderContext, renderArgs, &background);
-            if (background == nullptr)
-            {
-                return;
-            }
-        }
+        ComPtr<IImageSource> bitmapImageSource;
+        THROW_IF_FAILED(bitmapImage.As(&bitmapImageSource));
+
+        ComPtr<IUIElement> imageAsUIElement;
+        THROW_IF_FAILED(image.As(&imageAsUIElement));
+        THROW_IF_FAILED(image->put_Source(bitmapImageSource.Get()));
+        THROW_IF_FAILED(image.CopyTo(backgroundImage));
+    }
+
+    void ApplyBackgroundToRoot(_In_ IPanel* rootPanel,
+                               _In_ IAdaptiveBackgroundImage* backgroundImage,
+                               _In_ IAdaptiveRenderContext* renderContext)
+    {
+        HString url;
+        THROW_IF_FAILED(backgroundImage->get_Url(url.GetAddressOf()));
 
         ComPtr<IImage> xamlImage;
-        THROW_IF_FAILED(background.As(&xamlImage));
+        CreateBackgroundImage(renderContext, url.Get(), &xamlImage);
+        if (!xamlImage)
+        {
+            return;
+        }
 
-        ABI::AdaptiveCards::ObjectModel::Uwp::BackgroundImageFillMode fillMode;
-        THROW_IF_FAILED(backgroundImage->get_FillMode(&fillMode));
```

Comment:
this is just simple dead code, yes? have you verified that [`fillMode`](https://adaptivecards.io/explorer/BackgroundImage.html) still works?

Reply:
**fillMode** is being applied in **TileControl**, checking now..